core/modules/user/user.module

```diff
@@ -1835,11 +1835,14 @@ function user_admin_paths() {
  * Implements hook_user_login_paths().
  */
 function user_user_login_paths() {
-  return array(
+  $paths = array(
     'user/login' => TRUE,
     'user/password' => TRUE,
-    'user/register' => TRUE,
   );
+  if (config_get('system.core', 'user_register') != USER_REGISTER_ADMINISTRATORS_ONLY) {
```

Comment:
In my testing with a clean install (no TFA or other modules) this doesn't seem necessary, now that we are catching this in `user_custom_theme()`. In fact if I revert this part of the PR, the fix continues to work just fine.

Reply:
It's catching the problem early. Not strictly necessary, you're right, but why adding the user/register path, if we already know, it will never get the simplified display, as it's 403 then, anyway. :wink: